gateway/src/main/java/org/georchestra/gateway/filter/headers/RemoveSecurityHeadersGatewayFilterFactory.java

```diff
@@ -49,12 +50,13 @@ public class RemoveSecurityHeadersGatewayFilterFactory extends AbstractGatewayFi
 
     private static final String DEFAULT_SEC_HEADERS_PATTERN = "(?i)(sec-.*|Authorization)";
 
-    private final RemoveHeadersGatewayFilterFactory delegate = new RemoveHeadersGatewayFilterFactory();
+    private final RemoveHeadersGatewayFilterFactory delegate;
     private final RemoveHeadersGatewayFilterFactory.RegExConfig config = new RemoveHeadersGatewayFilterFactory.RegExConfig(
             DEFAULT_SEC_HEADERS_PATTERN);
 
-    public RemoveSecurityHeadersGatewayFilterFactory() {
+    public RemoveSecurityHeadersGatewayFilterFactory(GatewayConfigProperties configProps) {
         super(Object.class);
+        delegate = new RemoveHeadersGatewayFilterFactory(configProps);
     }
```

Comment:
same question, I'd expect that we could `@Autowired` it

Reply:
I'll take a look at all those comments about `@Autowired`, but I don't like autowired at all, prefer constructor injection and resolving the dependencies in the `@AutoConfiguration` classes. Makes it clearer/cleaner and easier to test.

Reply:
> but I don't like autowired at all

I thought this was the way to go, I won't object and trust you if you don't want to use them.